### PR TITLE
Add some memory assert_invalid and dummy control flow

### PR DESCRIPTION
**Pull request by andreaTP**
Partially targets #335

### CODE REVIEW

File: runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java [hunk 1]
```diff
@@ -217,6 +217,10 @@ public Instance initialize(boolean start) {
             }
         }
 
+        if (start && module.export(START_FUNCTION_NAME) != null) {
```
Comment (bhelx) [inline runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java:220]: Can you explain this change?

File: runtime-tests/pom.xml [hunk 2]
```diff
@@ -171,6 +171,7 @@
             <wast>func.wast</wast>
             <wast>func_ptrs.wast</wast>
             <wast>global.wast</wast>
+            <wast>i32.wast</wast>
```
Comment (electrum) [inline runtime-tests/pom.xml:174]: Is this a regression?

File: runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java (hunk 1 above)
Comment (electrum) [reply runtime/src/main/java/com/dylibso/chicory/runtime/Instance.java:220]: This means we are now doing validation before running the start function -- which seems correct.

File: runtime/src/main/java/com/dylibso/chicory/runtime/Module.java [hunk 3]
```diff
@@ -226,11 +226,9 @@ protected Instance instantiate(
 
         Memory memory = null;
         if (module.memorySection() != null) {
```
Comment (electrum) [inline runtime/src/main/java/com/dylibso/chicory/runtime/Module.java:228]: When we cleanup the APIs, we should consider changing all of these nullables to `Optional`

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/MemoryLimits.java [hunk 4]
```diff
@@ -63,14 +65,12 @@ public MemoryLimits(int initial, int maximum) {
      * @param shared {@code true} if the limits apply to a shared memory segment, or {@code false} otherwise
      */
     public MemoryLimits(int initial, int maximum, boolean shared) {
-        if (initial < 0 || initial > maximum) {
-            throw new IllegalArgumentException(
-                    "initial must be >= 0 and <= maximum, but was " + initial);
+        if (initial > MAX_PAGES || maximum > MAX_PAGES || initial < 0 || maximum < 0) {
```
Comment (electrum) [inline wasm/src/main/java/com/dylibso/chicory/wasm/types/MemoryLimits.java:68]: Could do a separate check for negative, unless this error message is required per specification.
```java
if (initial < 0 || maximum < 0) {
    throw new InvalidException("memory size is negative");
} else if (initial > MAX_PAGES || maximum > MAX_PAGES) {

**Review by electrum — APPROVED** [review]
Looks like a great start! I'm learning a lot by reading through this code.

File: runtime-tests/pom.xml (hunk 2 above)
Comment (andreaTP) [reply runtime-tests/pom.xml:174]: not really, just "more correct" checks, expect a bit of forth and back until we have a super solid control flow.

File: runtime/src/main/java/com/dylibso/chicory/runtime/Module.java (hunk 3 above)
Comment (andreaTP) [reply runtime/src/main/java/com/dylibso/chicory/runtime/Module.java:228]: Sadly `Optional`s are a leaky abstraction in Java, I would be happy to discuss this change, I might be overthinking something.

File: wasm/src/main/java/com/dylibso/chicory/wasm/types/MemoryLimits.java (hunk 4 above)
Comment (andreaTP) [reply wasm/src/main/java/com/dylibso/chicory/wasm/types/MemoryLimits.java:68]: > unless this error message is required per specification.

It's the case, for context `< 0` usually happens during overflows other than real negative input passed by the user.

**Comment by andreaTP** [comment]
I merge this as a checkpoint, will iterate on it, feel free to open separate issues if I miss/forget to follow up on something required!

**Pull request merged**